src/backend/aspen/api/settings.py

```diff
@@ -270,3 +226,58 @@ def customise_sources(
                 aws_secret_settings,
                 aws_ssm_settings,
             )
+
+
+class APISettings(Settings):
+    """Pydantic Settings object - do not instantiate it directly, please use get_settings() as a dependency where possible"""
+
+    # Env vars usually read from env vars
+    FLASK_ENV: str
+    API_URL: str
+
+    ####################################################################################
+    # Stack name
+    @cached_property
+    def AUTH0_LOGOUT_URL(self) -> str:
+        flask_env = self.FLASK_ENV
+        if flask_env == "production":
+            return f"{self.AUTH0_BASE_URL}/v2/logout"
+        else:
+            return f"https://{self.AUTH0_DOMAIN}/Account/Logout"
+
+    @cached_property
+    def AUTH0_CALLBACK_URL(self) -> str:
+        api_url = self.API_URL
+        return f"{api_url}/v2/auth/callback"
+
+    @cached_property
+    def AUTH0_BASE_URL(self) -> str:
+        if self.SECRET_AUTH0_BASE_URL:
+            return self.SECRET_AUTH0_BASE_URL
+        else:
+            return f"https://{self.AUTH0_DOMAIN}"
+
+    @cached_property
+    def AUTH0_ACCESS_TOKEN_URL(self) -> str:
+        if self.SECRET_AUTH0_ACCESS_TOKEN_URL:
+            return self.SECRET_AUTH0_ACCESS_TOKEN_URL
+        else:
+            return f"{self.AUTH0_BASE_URL}/oauth/token"
+
+    @cached_property
+    def AUTH0_AUTHORIZE_URL(self) -> str:
+        if self.SECRET_AUTH0_AUTHORIZE_URL:
+            return self.SECRET_AUTH0_AUTHORIZE_URL
+        else:
+            return f"{self.AUTH0_BASE_URL}/authorize"
+
+    @property
+    def AUTH0_SERVER_METADATA_URL(self) -> str:
+        if self.SECRET_AUTH0_SERVER_METADATA_URL:
+            return self.SECRET_AUTH0_SERVER_METADATA_URL
+        else:
+            return f"https://{self.AUTH0_DOMAIN}/.well-known/openid-configuration"
+
+
+class CLISettings(Settings):
```

Comment:
The API service requires several environment variables to be set, that we currently don't (and don't have any reason to) send to batch jobs that run CLI commands. The refactoring of Settings / APISettings / CLISettings is just to get us to the point where API and CLI settings work **similarly** but with a few differences.